lib/elsa/consumer/worker.ex

```diff
@@ -58,11 +58,11 @@ defmodule Elsa.Group.Worker do
       connection: Keyword.fetch!(init_args, :connection),
       topic: Keyword.fetch!(init_args, :topic),
       partition: Keyword.fetch!(init_args, :partition),
-      generation_id: Keyword.fetch!(init_args, :generation_id),
+      generation_id: Keyword.get(init_args, :generation_id),
```

Comment:
are you allowing this to be `nil` if it's not enforcing a required value?

Reply:
Yeah I want it to be `nil` to denote that this consumer is not attached to a group. A group consumer will always have generation id and a simple consumer will never have one, AFAIK.